### DISCUSSION TITLE
Access actix websocket actor

### DISCUSSION THREAD

**Discussion by Sytten** [async-graphql/async-graphql#1104] [Q&A]
Hi!

I am implementing authentication in our graphql server and I am trying to find a way to close websocket connections when their token is expired. Ideally I would like to:
1. Check on each new message if the session is still alive (aka a callback on message), and be able to close in the callback
2. Close it from an external perspective elsewhere in the program (like if the user hits a logout endpoint). That could be done with an rx/tx send to the callback function.

I looked around the code a bit but I didn't find anything that could help me accomplish that.
Thanks!

**Comment by ruohki** [comment]
Iam interested in this topic too. I got authentication working via guards on the the field level. On my GET/POST /graphql routes i authenticate the user based on an auth-header and append the user object to my graphq-request wich i pass into my .execute fn on the schema.

Howerver for the subscriptions i have a hard time figuring out where to hook in authentication. Ofc i can do all authentication stuff on the route that initialized the subscription however i dont see a way on how to pass my user into the schema context so that it can be used by my guard
